src/dashboards/components/DashboardContainer.tsx

```diff
@@ -46,25 +45,39 @@ const DashboardContainer: FC = () => {
   const isEditing = useRouteMatch(
     '/orgs/:orgID/dashboards/:dashboardID/cells/:cellID/edit'
   )
-  const registerListeners = useCallback(() => {
-    if (timer.current) {
-      registerStopListeners()
-    }
 
-    timer.current = setTimeout(() => {
-      dispatch(resetAutoRefresh(`dashboard-${dashboardID}`))
-      dispatch(notify(dashboardAutoRefreshTimeoutSuccess()))
-      registerStopListeners()
-      GlobalAutoRefresher.stopPolling()
-      event('dashboards.autorefresh.dashboardcontainer.inactivitytimeout', {
-        timeout: autoRefresh.inactivityTimeout,
-      })
-    }, autoRefresh.inactivityTimeout)
+  const startTimeout = () => {
+    GlobalAutoRefresher.startTimeout(
+      timeoutFunction,
+      autoRefresh.inactivityTimeout
+    )
+  }
 
+  const registerListeners = useCallback(() => {
     window.addEventListener('load', registerListeners)
     document.addEventListener('mousemove', registerListeners)
     document.addEventListener('keypress', registerListeners)
-  }, [dashboardID, autoRefresh.inactivityTimeout])
+  }, [])
+
+  const registerStopListeners = useCallback(() => {
```

Comment:
i dont understand why you made this distinction

Reply:
is this missing a globalautorefesher.stop or something?

Reply:
omg nm. my brain read these as the same two functions. sorry

Reply:
Moving it up or simplifying it? Moved it up because the timeoutFunction below was complaining about invoking a function prior to it being declared. Besides that I stripped it down so it just removes the listeners